dockstore-client/src/main/java/io/dockstore/client/cli/YamlVerifyUtility.java

```diff
@@ -0,0 +1,182 @@
+/*
+ *    Copyright 2022 OICR and UCSC
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package io.dockstore.client.cli;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.dockstore.common.yaml.DockstoreYaml12;
+import io.dockstore.common.yaml.DockstoreYamlHelper;
+import io.dockstore.common.yaml.Service12;
+import io.dockstore.common.yaml.YamlWorkflow;
+import org.yaml.snakeyaml.Yaml;
+
+import static io.dockstore.client.cli.ArgumentUtility.out;
+
+/*
+    GENERAL STEPS:
+    1. Verfiy that YAML syntax is valid
+    2. Verify that all fields are named correctly
+    3. Verify that all required fields are present
+    4. Verify that all files exist
+ */
+
+public final class YamlVerifyUtility {
+
+    public static final String DOCKSTOREYML = ".dockstore.yml";
+    public static final String YAML = "yaml";
+
+    public static final String VALID_DOCKSTORE_YML = " is a valid dockstore yaml file and all required files are present";
+    public static final String INVALID_DOCKSTORE_YML = "Your .dockstore.yml is invalid:\n";
+
+    public static final String CONTAINS_ERRORS = " has the following errors:\n";
+    public static final String EMPTY_FILE = " is empty";
+    public static final String FILE_DOES_NOT_EXIST = " does not exist";
+    public static final String INVALID_FILE_STRUCTURE = "Your file structure has the following errors:\n";
+
+    public static final String INVALID_YAML = " is not a valid " + YAML + "file:\n";
+    // This message is displayed when it is determined that DOCKSTOREYML is a valid yaml file,
+    // but displaying this message does NOT mean DOCKSTOREYML is a valid dockstore yaml file
+    public static final String VALID_YAML_ONLY = " is a valid " + YAML + " file";
+
+
+
+    private YamlVerifyUtility() {
+    }
+    // Determines if all the files referenced in a list of strings exist
+    private static List<String> filesExist(List<String> paths, String base) {
+        List<String> missingFiles = new ArrayList<>();
+        for (String path : paths) {
+            Path pathToFile = Paths.get(base, path);
+            if (!Files.exists(pathToFile)) {
+                missingFiles.add(pathToFile.toString() + FILE_DOES_NOT_EXIST);
```

Comment:
When I run this it detects an error like this:

```
/Users/charlesoverbeck/git/dockstore/dockstore-cli/dockstore-client/src/test/resources/YamlVerifyTestDirectory/2ToolsWithNoName/dockstore.wdl.json does not exist
```

I think it would be more useful to print out relative paths -- the user is working in the confined space of a repository. Just a suggestion, I could see an argument for the absolute path.

Reply:
This a feature I was attempting to have, currently what it is displaying is the directory you entered as the path parameter with the relate path in `.dockstore.yml` appended to it.

The issue is that if I use `~`  it appears that my system automatically resolves that to `/home/fhembroff` and then uses that as the command line parameter.

Here are two examples of the output when being run in `~/Documents/Dockstore/dockstore-cli` on my computer,
```
 java -jar dockstore-client/target/dockstore-client-1.13.0-SNAPSHOT-shaded.jar yaml --path ~/Documents/Dockstore/dockstore-cli/dockstore-client/src/test/resources/YamlVerifyTestDirectory/some-files-present validate

/home/fhembroff/Documents/Dockstore/dockstore-cli/dockstore-client/src/test/resources/YamlVerifyTestDirectory/some-files-present/.dockstore.yml is a valid yaml file
Your file structure has the following errors:
/home/fhembroff/Documents/Dockstore/dockstore-cli/dockstore-client/src/test/resources/YamlVerifyTestDirectory/some-files-present/dockstore.wdl.json does not exist
/home/fhembroff/Documents/Dockstore/dockstore-cli/dockstore-client/src/test/resources/YamlVerifyTestDirectory/some-files-present/Dockstore.cwl does not exist
```

```
java -jar dockstore-client/target/dockstore-client-1.13.0-SNAPSHOT-shaded.jar yaml --path dockstore-client/src/test/resources/YamlVerifyTestDirectory/some-files-present validate

dockstore-client/src/test/resources/YamlVerifyTestDirectory/some-files-present/.dockstore.yml is a valid yaml file
Your file structure has the following errors:
dockstore-client/src/test/resources/YamlVerifyTestDirectory/some-files-present/dockstore.wdl.json does not exist
dockstore-client/src/test/resources/YamlVerifyTestDirectory/some-files-present/Dockstore.cwl does not exist
```

Now given that most users will have the dockstore script in their paths I thought most of them would run this script in the directory of `.dockstore.yml`.

Here is an example of the output when the following is run in `~/Documents/Dockstore/dockstore-cli/dockstore-client/src/test/resources/YamlVerifyTestDirectory/some-files-present ` on my computer.

```
dockstore  yaml --path ./ validate

./.dockstore.yml is a valid yaml file
Your file structure has the following errors:
./dockstore.wdl.json does not exist
./Dockstore.cwl does not exist
```

Reply:
I think you could determine it out using `Paths.relativize` or something like that, regardless of how the path is specified. But looking at the Linux `find` command, and you're doing what they're doing, so all good -- sorry for the digression.

```
~/temp$ find . -name index.html
./gatling/1.12.0/index.html
./gatling/1.12.2/index.html
./gatling/1.12.0-1.12.2/index.html
~/temp$ find ~/temp -name index.html
/Users/charlesoverbeck/temp/gatling/1.12.0/index.html
/Users/charlesoverbeck/temp/gatling/1.12.2/index.html
/Users/charlesoverbeck/temp/gatling/1.12.0-1.12.2/index.html
```